### PR TITLE
Refactorizacion de la logica de redireccionamiento del menu

### PR DESCRIPTION
**Pull request by MarioRam0s**
## Description
Se refactorizo la interfaz de menu y la logica de redireccionamiento para que sea mas eficiente

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (no functional changes, no api changes)
- [ ] This change requires a documentation update
- [ ] Other (please describe): 

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in areas hard to understand.
- [ ] I have made the corresponding changes to the documentation.
- [x] My changes do not generate new warnings.
- [ ] I have added tests that prove my solution is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] All dependent changes have been merged and published in subsequent modules.
- [x] I have reviewed my code and corrected any spelling mistakes.

### CODE REVIEW

File: lib/generated/intl/messages_en.dart [hunk 1]
```diff
@@ -87,7 +87,7 @@ class MessageLookup extends MessageLookupByLibrary {
             MessageLookupByLibrary.simpleMessage(
                 "Debe ingresar el nombre de usuario o correo"),
         "Derechos_reservados": MessageLookupByLibrary.simpleMessage(
-            "© 2023 Hope. All rights reserved"),
+            "© 2024 Hope. All rights reserved"),
```
Comment (SamBarFu) [inline lib/generated/intl/messages_en.dart:90]: Que sea dinámico

File: lib/generated/intl/messages_es_NI.dart [hunk 2]
```diff
@@ -87,7 +87,7 @@ class MessageLookup extends MessageLookupByLibrary {
             MessageLookupByLibrary.simpleMessage(
                 "Debe ingresar el nombre de usuario o correo"),
         "Derechos_reservados": MessageLookupByLibrary.simpleMessage(
-            "© 2023 Hope. All rights reserved"),
+            "© 2024 Hope. All rights reserved"),
```
Comment (SamBarFu) [inline lib/generated/intl/messages_es_NI.dart:90]: Que sea dinámico

**Review by SamBarFu — CHANGES_REQUESTED** [review]

File: lib/generated/intl/messages_en.dart (hunk 1 above)
Comment (MarioRam0s) [reply lib/generated/intl/messages_en.dart:90]: este cambio se subira en el ultimo PR

File: lib/generated/intl/messages_es_NI.dart (hunk 2 above)
Comment (MarioRam0s) [reply lib/generated/intl/messages_es_NI.dart:90]: este cambio se subira en el ultimo PR

**Review by SamBarFu — APPROVED** [review]

**Pull request merged**